UnrecognizedPropertyException thrown for unexpected `date_created` attribute in `file_versions` serialization

java -jar target/osf-packager-cli-1.1.2-SNAPSHOT.jar -c ~/iCloud/osf-configs/osf-client.conf https://api.osf.io/v2/registrations/3gvg9/
There was an unrecoverable error: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "date_created" (class org.dataconservancy.cos.osf.client.model.FileVersion), not marked as ignorable (4 known properties: "content_type", "size", "id", "links"])
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.FileVersion["date_created"])
java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "date_created" (class org.dataconservancy.cos.osf.client.model.FileVersion), not marked as ignorable (4 known properties: "content_type", "size", "id", "links"])
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.FileVersion["date_created"])
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectCollectionInternal(ResourceConverter.java:283)
	at com.github.jasminb.jsonapi.ResourceConverter.unmarshalAndSetCollection(ResourceConverter.java:604)
	at com.github.jasminb.jsonapi.ResourceConverter.handleRelationships(ResourceConverter.java:491)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:316)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectCollectionInternal(ResourceConverter.java:263)
	at com.github.jasminb.jsonapi.ResourceConverter.unmarshalAndSetCollection(ResourceConverter.java:604)
	at com.github.jasminb.jsonapi.ResourceConverter.handleRelationships(ResourceConverter.java:491)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:316)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectCollectionInternal(ResourceConverter.java:263)
	at com.github.jasminb.jsonapi.ResourceConverter.unmarshalAndSetCollection(ResourceConverter.java:604)
	at com.github.jasminb.jsonapi.ResourceConverter.handleRelationships(ResourceConverter.java:491)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:316)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectCollectionInternal(ResourceConverter.java:263)
	at com.github.jasminb.jsonapi.ResourceConverter.unmarshalAndSetCollection(ResourceConverter.java:604)
	at com.github.jasminb.jsonapi.ResourceConverter.handleRelationships(ResourceConverter.java:491)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:316)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:208)
	at com.github.jasminb.jsonapi.ResourceConverter.readObject(ResourceConverter.java:184)
	at org.dataconservancy.cos.osf.client.retrofit.PaginatedResponseBodyConverter.convert(PaginatedResponseBodyConverter.java:84)
	at org.dataconservancy.cos.osf.client.retrofit.PaginatedResponseBodyConverter.convert(PaginatedResponseBodyConverter.java:31)
	at retrofit.OkHttpCall.parseResponse(OkHttpCall.java:148)
	at retrofit.OkHttpCall.execute(OkHttpCall.java:116)
	at org.dataconservancy.cos.packaging.cli.PackageGenerationApp.run(PackageGenerationApp.java:218)
	at org.dataconservancy.cos.packaging.cli.PackageGenerationApp.main(PackageGenerationApp.java:190)
Caused by: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "date_created" (class org.dataconservancy.cos.osf.client.model.FileVersion), not marked as ignorable (4 known properties: "content_type", "size", "id", "links"])
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.FileVersion["date_created"])
	at com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException.from(UnrecognizedPropertyException.java:62)
	at com.fasterxml.jackson.databind.DeserializationContext.reportUnknownProperty(DeserializationContext.java:855)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer.handleUnknownProperty(StdDeserializer.java:1083)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownProperty(BeanDeserializerBase.java:1389)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.handleUnknownVanilla(BeanDeserializerBase.java:1367)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:266)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:3760)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2042)
	at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:2529)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:304)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectCollectionInternal(ResourceConverter.java:263)
	... 23 more

--- a/osf-core/osf-core-model/src/main/java/org/dataconservancy/cos/osf/client/model/FileVersion.java
+++ b/osf-core/osf-core-model/src/main/java/org/dataconservancy/cos/osf/client/model/FileVersion.java
@@ -17,6 +17,7 @@ package org.dataconservancy.cos.osf.client.model;
 
 import java.util.Map;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.github.jasminb.jsonapi.annotations.Id;
 import com.github.jasminb.jsonapi.annotations.Links;
@@ -27,6 +28,8 @@ import com.github.jasminb.jsonapi.annotations.Type;
  *
  */
 @Type("file_versions")
+// see https://github.com/CenterForOpenScience/osf.io/issues/7516
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class FileVersion {
 
     @Id

--- a/osf-packager/osf-packager-model/src/test/java/org/dataconservancy/cos/osf/packaging/RegistrationPackageTest.java
+++ b/osf-packager/osf-packager-model/src/test/java/org/dataconservancy/cos/osf/packaging/RegistrationPackageTest.java
@@ -464,7 +464,7 @@ public class RegistrationPackageTest extends AbstractMockServerTest {
         // Move of RegistrationMeta to a Map<String, RegistrationMeta> "hides" the RegistrationMeta class from
         // annotation processing.
         // TODO: https://github.com/DataConservancy/dcs-packaging-osf/issues/4
-        assertEquals(94, result.size());
+        assertEquals(95, result.size());
 
         final AnnotatedElementPair aep1 = new AnnotatedElementPair(r.getClass(), OwlIndividual.class);
         final AnnotatedElementPair aep2 = new AnnotatedElementPair(r.getClass(), OwlIndividual.class);

